src/zmalloc.c

```diff
@@ -93,7 +98,7 @@ static void (*zmalloc_oom_handler)(size_t) = zmalloc_default_oom;
  * '*usable' is set to the usable size if non NULL. */
 void *ztrymalloc_usable(size_t size, size_t *usable) {
     ASSERT_NO_SIZE_OVERFLOW(size);
-    void *ptr = malloc(size+PREFIX_SIZE);
+    void *ptr = malloc(MALLOC_MIN_SIZE(size)+PREFIX_SIZE);
```

Comment:
since we do a `MALLOC_MIN_SIZE(size)` when doing zmalloc(0), do we also need to update update_zmalloc_stat_alloc below?

```diff
- update_zmalloc_stat_alloc(size);
- update_zmalloc_stat_alloc(size+PREFIX_SIZE);

+ update_zmalloc_stat_alloc(MALLOC_MIN_SIZE(size));
+ update_zmalloc_stat_alloc(MALLOC_MIN_SIZE(size)+PREFIX_SIZE);

Reply:
i suppose both are correct.

when malloc_size exists, we record the size that was allocated instead of the size that was requested.
this would work with both jemalloc, and libc malloc (the change in this PR doesn't break this.

when malloc_size is missing, we allocate extra size_t bytes and store the requested size in it.
in that case, the requested size is probably different than the allocated size anyway (the change in this PR doesn't conceptually change that).

do we have room for improvement since in this case we are aware of the extra bytes **we** asked for.
same as we're also aware of the extra size_t bytes we asked for.
so i suppose you can make a PR to "fix" this.